predict-spec-cost-api.md

```diff
@@ -0,0 +1,145 @@
+# Spec Cost Prediction API
+
+The `/model/prediction/speccost` API takes Kubernetes API Objects ("workloads")
+as input and produces a cost impact prediction for them, including a diff if a
+matching existing workload can be found in the cluster.
+
+Currently supported workload types:
+- Deployments
+- StatefulSets
+- Pods
+
+## Calling the API
+
+The endpoint accepts HTTP POST requests. It accepts the following query parameters:
+
+| Parameter | Type | Description |
+|-----------|------|-------------|
+| `clusterID` | string (required) | The Kubecost cluster ID of the cluster in which the workloads will be deployed. Currently, this must be the same as the cluster ID of the Kubecost installation which is serving the `/speccost` endpoint. Support for multi-cluster is planned. |
+| `defaultNamespace` | string (required) | The namespace in which namespace-scoped objects will be "deployed" to if no namespace is set in the standard metadata field of the object. |
+| `window` | string (default `2d`) | The Kubecost data window used for determining resource costs fed into the cost prediction. |
+| `noUsage` | bool (default `false`) | Set to `true` to ignore historical usage data (if it exists) when making the prediction. This is equivalent to making a prediction using only requests. |
+
+The API requires that workloads be passed in the request body in YAML format and
+that the `Content-Type` header be set to `application/yaml`. Multiple workloads
+can be passed via separation with the standard `---` syntax.
+
+### Example
```

Comment:
The example is nice. However, I think it would be easier for reader to follow if we break it into 2 parts:
- 1. create an example deployment manifest and pass it into env var
- 2. call  `/model/prediction/speccost` with curl to retrieve the cost prediction.

Reply:
Yes, that would probably be better -- or even write it to a file, then pass to curl with the `@` syntax (or whatever it is...). I just copy-pasted my testing block here, which is why its all one big nasty shell script.

I should be able to make an update soon.